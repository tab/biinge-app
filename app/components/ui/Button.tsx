import React from "react"
import { Pressable, Text, ViewStyle, TextStyle, StyleProp } from "react-native"

import { buttonStyles } from "styles"

type Props = {
  style?: StyleProp<ViewStyle>
  textStyle?: StyleProp<TextStyle>
  disabled?: boolean
  loading?: boolean
  testID?: string
  children: React.ReactNode
  onPress: () => void
}

const ButtonComponent = ({
  style,
  textStyle,
  disabled,
  loading,
  testID,
  children,
  onPress,
}: Props) => {
  return (
    <Pressable
      testID={testID}
      style={
        disabled || loading
          ? [buttonStyles.button, buttonStyles.disabled, style]
          : [buttonStyles.button, style]
      }
      disabled={disabled || loading}
      onPress={onPress}
    >
      <Text style={[buttonStyles.text, textStyle]}>{children}</Text>
    </Pressable>
  )
}

ButtonComponent.defaultProps = {
  disabled: false,
  loading: false,
}

export default ButtonComponent
