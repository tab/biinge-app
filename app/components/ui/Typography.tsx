import React from "react"
import { Text, StyleProp, TextStyle } from "react-native"

import { typographyStyles, TypographyVariantType } from "styles"

type Props = {
  style?: StyleProp<TextStyle>
  variant: TypographyVariantType
  numberOfLines?: number
  children: React.ReactNode
}

const TypographyComponent = ({
  style,
  variant,
  numberOfLines,
  children,
}: Props) => {
  const getVariantStyle = () => typographyStyles[variant] || {}

  return (
    <Text numberOfLines={numberOfLines} style={[getVariantStyle(), style]}>
      {children}
    </Text>
  )
}

TypographyComponent.defaultProps = {
  numberOfLines: 0,
}

export default TypographyComponent
