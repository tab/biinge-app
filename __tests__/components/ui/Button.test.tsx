import React from "react"
import "react-native"
import { render } from "@testing-library/react-native"

import { it } from "@jest/globals"

import Button from "components/ui/Button"
import Typography from "components/ui/Typography"

describe("Button", () => {
  const title = "Button title"
  const handleClick = jest.fn()

  it("renders disabled button", () => {
    const disabled = true

    const tree = render(
      <Button disabled={disabled} onPress={handleClick}>
        <Typography variant="body">{title}</Typography>
      </Button>,
    )
    expect(tree).toMatchSnapshot()
  })

  it("renders button", () => {
    const disabled = false

    const tree = render(
      <Button disabled={disabled} onPress={handleClick}>
        <Typography variant="body">{title}</Typography>
      </Button>,
    )
    expect(tree).toMatchSnapshot()
  })
})
