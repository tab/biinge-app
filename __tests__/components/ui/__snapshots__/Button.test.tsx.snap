// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders button 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 50,
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 15,
      },
      undefined,
    ]
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={0}
    style={
      [
        {
          "color": "#181818",
          "fontSize": 18,
          "fontWeight": "300",
          "lineHeight": 22,
        },
        undefined,
      ]
    }
  >
    Button title
  </Text>
</View>
`;

exports[`Button renders disabled button 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 50,
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 15,
      },
      {
        "opacity": 0.5,
      },
      undefined,
    ]
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={0}
    style={
      [
        {
          "color": "#181818",
          "fontSize": 18,
          "fontWeight": "300",
          "lineHeight": 22,
        },
        undefined,
      ]
    }
  >
    Button title
  </Text>
</View>
`;
